android/titanium/src/java/org/appcelerator/titanium/proxy/TiViewProxy.java

```diff
@@ -737,7 +737,32 @@ public void removeAllChildren()
 			}
 		}
 	}
+	
+	/**
+	* Returns the view by the given ID.
+	* @module.api
+	*/
+	@Kroll.method
+	public TiViewProxy getViewById(String arg)
+	{
+		if (children != null) {
+			for (TiViewProxy child : children) {
+				if (child.children != null && child.children.size() > 0) {
+					TiViewProxy parentChild = child.getViewById(arg);
+					if (parentChild != null) {
+						return parentChild;
+					}
+				}
+
+				if (child.getProperty(TiC.PROPERTY_ID) && child.getProperty(TiC.PROPERTY_ID).equals(arg)) {
```

Comment:
But that's how it was [before](https://github.com/appcelerator/titanium_mobile/pull/8677/commits/5df4ff7b1845481d17a5e21e2d1c5384ce8a2f8b#diff-bec22e2efdd8b352a7754ea42f435dc7L757) all that. You can edit it on Github, so we can ensure it doesn't cause problems again.